src/main/java/uk/gov/companieshouse/api/strikeoffobjections/config/ValidationConfig.java

```diff
@@ -0,0 +1,37 @@
+package uk.gov.companieshouse.api.strikeoffobjections.config;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import uk.gov.companieshouse.api.strikeoffobjections.common.ApiLogger;
+import uk.gov.companieshouse.api.strikeoffobjections.model.entity.ObjectionStatus;
+import uk.gov.companieshouse.api.strikeoffobjections.validation.AllowedValuesValidationRule;
+import uk.gov.companieshouse.api.strikeoffobjections.validation.DisallowedValuesValidationRule;
+import uk.gov.companieshouse.api.strikeoffobjections.validation.ValidationRule;
+
+import java.util.Arrays;
+import java.util.List;
+
+@Configuration
+public class ValidationConfig {
+
+    @Value("#{'${ACTION_CODES_COMPANY_STRUCK_OFF}'.split(',')}")
+    private List<Long> companyStruckOffActionCodes;
+
+    @Value("#{'${ACTION_CODES_STRIKE_OFF_NOTICE}'.split(',')}")
+    private List<Long> strikeOffNoticeActionCodes;
+
+    @Bean
+    public List<ValidationRule<Long>> getActionCodeValidationRules(ApiLogger apiLogger) {
+        return Arrays.asList(
+                new DisallowedValuesValidationRule<>(
+                        companyStruckOffActionCodes,
+                        ObjectionStatus.INELIGIBLE_COMPANY_STRUCK_OFF,
+                        apiLogger),
+                new AllowedValuesValidationRule<>(
```

Comment:
This rule (or the name of it) seems odd. Why would the 'no current dissolution' action codes be assigned to an 'allowed' rule?

Reply:
The rule will check if the input value is in a list of allowable values. So if the action code is not in the allowed list, it will throw a validation exception with that status. The requirement was to check if the code was not one of the list.

Reply:
maybe i could rename them to DisallowedValue (no s) and same for the other one

Reply:
Understand this now...